_docs/3x/execution/scheduler.md

```diff
@@ -300,16 +300,20 @@ testScheduler.execute(runnable1)
 testScheduler.execute(runnable2)
 ```
 
+Note that the Execution model (explained below) still matters for a TestScheduler.
```

Comment:
I feel like it's not strictly about `ExecutionModel` but about a `Task` being possibly synchronous in which case we could change it to be asynchronous and one of the option is to use `AlwaysAsyncExecution` (we could also use executeAsync if we are only bothered about the start of execution).

Maybe we could add a small example with `Task { println("Immediate!") }` + `runToFuture` and provide a solution?

Reply:
True. I've reworded, what do you think?

I wasn't sure what a good example would look like, you're welcome to add to this PR if you have one in mind :)